tests/RobotFramework/tests/mqtt/disconnect_from_broker.robot

```diff
@@ -0,0 +1,56 @@
+*** Settings ***
+Resource        ../../resources/common.resource
+Library         ThinEdgeIO
+Library         OperatingSystem
+Library         String
+
+Suite Setup     Custom Setup
+Suite Teardown  Get Logs
+
+Test Tags      theme:mqtt
+
+
+*** Variables ***
+
+${TOPIC}             test/topic
+${MESSAGE}           Hello
+
+
+*** Test Cases ***
+
+Validate no Error message in journalctl
+    [Documentation]    This Test step is inspired by 'tedge mqtt cli commands do not disconnect cleanly from MQTT broker #2922' bug
+    ...                extended to check for any possible error that may appear in the journalctl
+    Execute Command    tedge mqtt pub foo bar
+    ${log}    Execute Command    journalctl -u mosquitto -n 30
+    Should Not Contain    ${log}    Error
+
+Publish Connection Closed On Interruption
+    [Documentation]    Tests that a publish connection to MQTT broker closes upon interruption.
+    Execute Command    tedge mqtt pub ${TOPIC} ${MESSAGE}    ignore_exit_code=True
```

Comment:
We don't need the ignore_exit_code flag for this since weren't run using `timeout` to kill the process.

```suggestion
    Execute Command    tedge mqtt pub ${TOPIC} ${MESSAGE}
```